### PR TITLE
Simplified implementation of various date formats.

### PR DESCRIPTION
**Pull request by ngnpope**
Split out from https://github.com/django/django/pull/15405 according to the plan in https://github.com/django/django/pull/15405#issuecomment-1295325606.

### CODE REVIEW

**Comment by ngnpope** [comment]
Flaky test failure unrelated.

File: django/utils/dateformat.py [hunk 1]
```diff
@@ -265,16 +265,13 @@ def o(self):
...
-            )
-        if is_naive(self.data):
-            dt = make_aware(self.data, timezone=self.timezone)
-        else:
-            dt = self.data
-        return format_datetime_rfc5322(dt)
+        value = self.data
+        if not isinstance(value, datetime):
+            # Assume midnight UTC if instance of datetime.date provided.
+            value = datetime.combine(value, time.min).replace(tzinfo=timezone.utc)
```
Comment (felixxm) [inline django/utils/dateformat.py:269]: Why not the `get_default_timezone()` instead? :thinking:

Comment (ngnpope) [reply django/utils/dateformat.py:269]: Think I just tend to default to UTC. But, you're right, default timezone would make more sense here.

Also checked how PHP behaves given that this implementation is largely based on theirs and the following:
```php
<?php
date_default_timezone_set('Australia/Sydney');
echo(date("r", 0));
```
Yields the following:
```
Thu, 01 Jan 1970 10:00:00 +1000
```
Will switch to use `get_default_timezone()` 👍🏻

Comment (ngnpope) [reply django/utils/dateformat.py]: Updated.

**Comment by felixxm** [comment]
@ngnpope Thanks :+1: I re-organized commits.

**Pull request merged**